iib/web/models.py

```diff
@@ -257,6 +257,7 @@ def get_or_create(cls, pull_specification: str) -> Image:
         if not image:
             image = Image(pull_specification=pull_specification)
             db.session.add(image)
+            db.session.flush()
```

Comment:
Question for discussion might be if it is OK to flush db.session at this place?
Does anybody see potential failure doing it?

Reply:
Why we are not doing that also after adding a User?

Reply:
even though users are added hardly ever you are right. I added flush there as well.

Reply:
> Question for discussion might be if it is OK to flush db.session at this place?
> Does anybody see potential failure doing it?

I don't see a problem with this. Looking at the [docs](https://docs.sqlalchemy.org/en/20/orm/session_basics.html#flushing) it would be eventually flushed anyway whenever a query would be executed, so doing this beforehand to avoid the aforementioned unique violation seems ok. It should also be an atomic operation as we're dealing with a postgreSQL in the backend so it should lock the operation before allowing a secondary "parallel flush" to happen.

It may impact a bit in performance, though, but I think it's something worthy in the tradeoff.

Reply:
It looked like adding db.session.flush() could fix the image uniqueviolation error first, but when digging deeper into this, I noticed we already create the session object with autoflush=True, shouldn't that take care of flushing the image object already?

Reply:
Thanks for the comment and I guess you are right in some way. From what I read in docu flush() will not help us here. Hence I change it to commit(). Flush will move the SQL queries to transaction buffer in DB. And it will stay there until commit is called.